.circleci/src/jobs/check_web.yml

```diff
@@ -10,16 +10,15 @@ steps:
   - checkout
   - check_config
   - restore_yarn_cache
-  - run:
-      name: Check linting
-      command: yarn workspace web lint
+  - lint:
+      directory: web
   - run:
       name: Check flow
       command: yarn workspace web flow:check-now --max-workers ${TOTAL_CPUS}
-  - run:
-      name: Unit tests
-      command: yarn workspace web test --ci --maxWorkers ${TOTAL_CPUS}
-  - run:
-      name: Test coverage # Results are not used right now. Run it anyway to (possibly) discover more problems.
-      command: yarn workspace web test:coverage --ci --maxWorkers ${TOTAL_CPUS}
+  - unit_test:
+      directory: web
+  - lint:
+      directory: api-client
+  - unit_test:
+      directory: api-client
```

Comment:
I'd rather just include api-client tests in both the web and native test suite directly via the roots config option: https://jestjs.io/docs/en/configuration.html#roots-arraystring

Advantages:
* No need to have jest setup files, configs scripts and dependencies in the api-client
* All relevant tests are executed when I run native/web tests locally
* The api-client tests belong to both or neither to web or native -> I don't see a reason why we'd execute api-client tests here but not in native
* Only babel-runtime is required in the api-client package

Reply:
I'm not a 100% sure what is the best solution. It could also be strange to execute the tests for the api-client with two different babel configs. But maybe we can not avoid this because metro and webpack will use different babel-configs nonetheless.

I'm also not sure which flow-config is used in the api-client. Again there are two flow-configs available and there could be differences.

Reply:
But I would be fine with your solution. I just dislike a bit how we manage a monorepo. The packages in there are not really packages which could life on its own. @Schedulaar what do you think?

Reply:
My gut feeling tells me, that it would be better to have a separate test environment for the api client.
Probably that feeling is based upon the benefits of modularity (i.e. if the api client is a separate package, we should treat it like that)

Also usually, if I run the tests for the webapp, because I changed something there, I'm not interested in the tests of the api client; vice versa, if I only make changes to the api client, I only want to run the tests for the api client.
(Imo, it makes sense to apply changes only to the api client: E.g. if new endpoints are introduced, there should probably be an issue first, that only adds the endpoint (and corresponding tests) to the api-client)

But this is nevertheless just a feeling :D Of course, as there are different babel and flow configs in web- and native app, testing the api-client with both gives the benefit of testing these configuration / glue code as well, but the tests in the api-client should not depend on that (optimally :sweat_smile: )

Reply:
I guess then we should have a separate circle ci job for api-client test though :thinking: Also, how does flow and linting fit into this? Should be made api-client specific as well then, right?

Reply:
I think when going the route of treating the api-client as separate module then we loose the benefits of a monorepo. This kind of sucks but I think we should go the easier route which probably works "well enought"®

Reply:
@maxammann which is the easier route which works well enough for you?

Reply:
Oke, there is the first problem with not handling module separation seriously.. I'm unable to collect the coverage results from the api-client because Jest only looks at the coverage results in the `rootDir` and not `roots`

Reply:
@klinzo I think this is a topic worth discussing, there are pro and cons on both sides

* Option A: No separation -> easy and hacky
* Option B: Total separation -> very hard to make it usable and not stress developers

I'd love to find a solution inbetween

Reply:
Should we discuss this in the call today?

Reply:
We will use a separate config